src/main/java/seedu/address/model/addresses/PublicAddressesComposition.java

```diff
@@ -0,0 +1,257 @@
+        return new PublicAddressesComposition(updatedPublicAddresses);
+    }
+
+    /**
+     * Checks if the composition is empty.
+     *
+     * @return True if the composition is empty, false otherwise.
+     */
+    public Boolean isEmpty() {
+        return publicAddresses.isEmpty();
+    }
+
+    /**
+     * Returns a string representation of the public addresses with indentation.
+     *
+     * @return A string representation of the public addresses.
+     */
+    public String toStringIndented() {
+        return publicAddresses.entrySet().stream().map(entry -> entry.getKey() + "\n" + INDENT
+                + INDENT + entry.getValue().stream().map(publicAddress -> publicAddress.getLabel() + ": "
+                        + publicAddress.getPublicAddressString())
+                .reduce((a, b) -> a + "\n" + b).orElse("")).reduce((a, b) -> a + "\n" + b).orElse("");
+    }
+
+    /**
+     * Creates a copy of the PublicAddressesComposition.
+     *
+     * @return A new PublicAddressesComposition with the same public addresses.
+     */
+    public PublicAddressesComposition copy() {
+        return new PublicAddressesComposition(publicAddresses);
+    }
+
+    /**
+     * Returns the number of networks in the composition.
+     *
+     * @return The number of networks.
+     */
+    public Integer size() {
+        return publicAddresses.size();
+    }
+
+    /**
+     * Returns the total number of public addresses in all networks.
+     *
+     * @return The total number of public addresses.
+     */
+    public Integer sizeOfAllPublicAddresses() {
+        return publicAddresses.values().stream().map(Set::size).reduce(0, Integer::sum);
+    }
+
+    /**
+     * Checks if a public address string exists among all networks.
+     *
+     * @param publicAddress The public address to check.
+     * @return True if the public address string exists, false otherwise.
+     */
+    public Boolean isPublicAddressStringAmongAllNetworks(PublicAddress publicAddress) {
+        return publicAddresses.values().stream()
+                .flatMap(Set::stream)
+                .anyMatch(pa -> pa.isPublicAddressStringEquals(publicAddress.getPublicAddressString()));
+    }
+
+    /**
+     * Returns the first public address in the composition.
+     *
+     * @return The first public address, or null if the composition is empty.
+     */
+    public PublicAddress getFirstPublicAddress() {
+        // publicAddresses map should not be null
+        assert publicAddresses != null;
+        // map should not be empty
+        // assert !publicAddresses.isEmpty();
+
+        // each set should not be empty
+        assert !publicAddresses.values().stream().anyMatch(Set::isEmpty);
+
+        // flatten map to get first public address
+        PublicAddress publicAddress =
+                publicAddresses.values().stream().flatMap(Set::stream).findFirst().orElse(null);
+        // get first public address
+        return publicAddress;
+    }
```

Comment:
Is this necessary? Seems like a bad idea since HashMap and HashSet are not ordered.

Reply:
the use case for this is to get any address and check if it exists in another set of addresses, i shall rename this function